packages/common/src/common.ts

```diff
@@ -60,7 +60,7 @@ export class Common {
   constructor(opts: CommonOpts) {
     this.events = new EventEmitter()
 
-    this._chainParams = JSON.parse(JSON.stringify(opts.chain))
+    this._chainParams = opts.chain
```

Comment:
This `JSON.parse(...` stuff was to do a deep copy of the original structure, I otherwise had the cases where the original data got modified when being modified in the `Common` object.

So, this e.g. concretely played out somewhere in the EVM/VM tests where a test broke when simply `Mainnet` or so was used as in input two times in a row to a `Common` object and in the first round the associated `_chainParams` got modified.

(I am doing a commit-by-commit review, bear with me if some of the stuff is getting fixed on later commits)

Reply:
( @jochem-brouwer even mentioned that we should do this in yet another place, can't ad hoc remember where (maybe along the `copy()` method))

Reply:
Btw, I learned just this week there is a better way to deep copy an object in js called structuredClone https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

Reply:
That'll be a nice thing to switch to if/when NodeJS brings it out from behind the `experimental` flag.  Looks like it's in Node 22 but not fully supported yet. https://nodejs.org/api/globals.html#structuredclonevalue-options